File: output/amazon_q/php/hard/q40-median-of-two-sorted-arrays.php
<?php
/** https://leetcode.com/problems/median-of-two-sorted-arrays */
// Problem: Median of Two Sorted Arrays

//Problem Description:
//Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
//The overall run time complexity should be O(log (m+n)).

// Constraints:
// nums1.length == m
// nums2.length == n
// 0 <= m <= 1000
// 0 <= n <= 1000
// 1 <= m + n <= 2000
// -10^6 <= nums1[i], nums2[i] <= 10^6

//Code Structure:

class Solution {

    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Float
     */
    function findMedianSortedArrays($nums1, $nums2) {
        $merged = array_merge($nums1, $nums2);
        sort($merged);
        $count = count($merged);
        if ($count % 2 == 0) {
            $middle = $count / 2;
            return ($merged[$middle - 1] + $merged[$middle]) / 2;
        } else {
            $middle = floor($count / 2);
            return $merged[$middle];
        }
    }
}
    

File: output/amazon_q/python/hard/q40-median-of-two-sorted-arrays.py
# https://leetcode.com/problems/median-of-two-sorted-arrays/
# Problem: Median of Two Sorted Arrays

#Problem Description:
#Given two sorted arrays nums1 and nums2 of size m and n respectively, return the median of the two sorted arrays.
#The overall run time complexity should be O(log (m+n)).

# Constraints:
# nums1.length == m
# nums2.length == n
# 0 <= m <= 1000
# 0 <= n <= 1000
# 1 <= m + n <= 2000
# -10^6 <= nums1[i], nums2[i] <= 10^6

#Code Structure:

class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        merged = []
        i = j = 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                merged.append(nums1[i])
                i += 1
            else:
                merged.append(nums2[j])
                j += 1
        merged.extend(nums1[i:])
        merged.extend(nums2[j:])
        total = len(merged)
        if total % 2 == 0:
            return (merged[total // 2 - 1] + merged[total // 2]) / 2.0
        else:
            return merged[total // 2]                                    
    